frb_codegen/src/ir/func.rs

```diff
@@ -6,7 +6,7 @@ pub struct IrFunc {
     pub inputs: Vec<IrField>,
     pub output: IrType,
     pub fallible: bool,
-    pub mode: IrFuncMode,
+    pub mode: (IrFuncMode, Option<usize>),
```

Comment:
What about creating a struct instead of this tuple, which will confuse code readers "what is this second arg?"

More specifically:

change

```rust
pub enum IrFuncMode {
    Normal,
    Sync,
    Stream,
}
```

to

```rust
pub enum IrFuncMode {
    Normal,
    Sync,
    Stream { some_field_name: usize },
}
```